sdk/azidentity/default_token_credential.go

```diff
@@ -41,15 +31,17 @@ func NewDefaultTokenCredential(options *DefaultTokenCredentialOptions) (*Chained
 	if !options.ExcludeEnvironmentCredential {
 		envCred, err := NewEnvironmentCredential(nil)
 		if err == nil {
-			cred.sources = append(cred.sources, envCred)
-			return NewChainedTokenCredential(cred.sources...)
+			creds = append(creds, envCred)
 		}
+		errMsg += "Make sure you have set the environment variables necessary for authentication: AZURE_TENANT_ID, AZURE_CLIENT_ID, AZURE_CLIENT_SECRET."
```

Comment:
Is this error message redundant?  Seems like it should be returned from `NewEnvironmentCredential()`.

Reply:
Yes, I agree I'll simply append err.Error() to errMsg